File: cpp/findMaxK.cpp
#include <unordered_set>
#include <vector>
#include <algorithm>
#include <cmath>

using namespace std;
class Solution {
public:
    int findMaxK(vector<int>& nums) {
    
        unordered_set<int> s;
        int m = -1;

        for (int i : nums) {
            s.insert(i);
            if (s.find(-i) !=s.end()) m = max(m, abs(i));
        }
        return m;
    }
};

File: rs/findMaxK.rs
use std::collections::HashSet;

impl Solution {
    pub fn find_max_k(nums: Vec<i32>) -> i32 {
        let mut s:HashSet<i32> = HashSet::new();
        let mut m:i32 = -1;

        for i in 0..nums.len() {
            s.insert(nums[i]);
            if s.contains(&-nums[i]) {
                m = m.max(nums[i].abs());
            }
        }
        return m;
    }
}